### DISCUSSION TITLE
Inflation

### DISCUSSION THREAD

**Discussion by andythenorth** [OpenTTD/OpenTTD#8429] [Features]
[Moved from #7589 to avoid derailing a specific PR]

See also #8427

**Comment by andythenorth** [comment]
Modest proposal: remove inflation. 

Inflation:
* makes assumptions about span of gameplay (start / end year) that don't hold in OpenTTD with variable start / end years
* makes the game borderline unplayable when inflation has reached a high level
* causes bug reports to newgrf authors about unplayable costs (I've been on both ends of this, filing useless bug reports on newgrfs due to inflation, and receiving them myself)

If we want increasing vehicle run costs over time, newgrf authors can take care of that themselves.

**Reply by ldpl** [reply]
Also inflation makes no gameplay difference for players that build stuff that can survive 170 years of it from the getgo. But it does annoy them to no end by scrambling familiar costs of stuff into a random junk.
And players that don't build like that get sneakily murdered when the time comes and have to restart.

Basically, the entirety of inflation gameplay is: you restart until it doesn't matter.

**Comment by michicc** [comment]
Modest counter proposal: fix inflation period (as per PR  #7589)

- variable start / end year doesn't matter anymore, as the inflation period is always the same
- costs can be balanced for one specific scenario, making balancing with inflation on practically possible
- costs are reproducible, thus lowering bug reports

**Comment by James103** [comment]
For those people that use inflation to slowly devalue the amount of currency they're holding, a Game Script could be used to instead remove a percentage of the company's bank balance and/or value every year.

**Reply by ldpl** [reply]
For people who hold currency, it's meaningless regardless of the amount. If they had anything to spend it on they would've done it already.

**Comment by JGRennison** [comment]
To throw another opinion in, it is the divergence between the income scaling factor and the cost scaling factor which is the problem, rather than the inflation of prices/costs per se.

If inflation inflated both scaling factors equally (or if the divergence was less severe), the unplayability and NewGRF balancing issues would go away, but difficulty would still be slightly increased relative to not having inflation, as the value of the giant money mountains accumulated by players would be gradually eroded over time.

**Reply by James103** [reply]
In other words, maybe split the "Interest rate" setting into 2 or 3 settings:
* Loan interest rate
* Cost inflation rate
* Income inflation rate

This would allow loan interest to be very high (or non-existent) compared to inflation, but it would also allow the divergence to be controlled or even reversed by the player (and/or Game Script).

Default values on load: N%, N%, and (N-1)% respectively, where N is the old interest rate.

See JGRennison/OpenTTD-patches#209

**Comment by andythenorth** [comment]
Modest counter counter counter counter proposal.

Newgrf entity 'economy' (or such - names are not important at this stage I hope).

**Monthly callback for setting a factor on base costs**

* global, affects whole map
* affects all costs: vehicles, construction, maintenance etc
* probably returns a multiplier to the base costs? (my implementation ideas are usually wrong tbh)

**Monthly callback for setting a factor on base income**

* global, affects whole map
* affects all income: delivering cargo, scrapping rails etc
* probably returns a multiplier to the base costs, and to cargo payments? (my implementation ideas are usually wrong tbh)

**Equivalent callbacks for setting base costs / income, but called when map gen completes**

* same behaviour, but intent is to adjust economy at game start
* (inspired by the cb that industries can optionally use when constructed)

**Monthly callbacks for setting base costs / income, but scoped per town**

* intent is to allow regional variation of both costs and income
* in principle this could represent things like
  * per cargo 'demand' via payment rates
  * local labour costs
  * incentives or disincentives to construction (high or low construction costs)
* this may not make any sense for, e.g. vehicle running costs, unless vehicles were given some concept of a home depot or home station, which I think extends the simulation too far into complexity
* implementation might be as a modifier to *current* global base costs? As this would be the simplest way to let the global script handle, e.g. inflation, recession etc, whilst varying result locally per town

**Comment by 0xBEEF7AC0** [comment]
Wouldn't removal of inflation go against the stated goal of OpenTTD? "...attempts to mimic the original game as closely as possible while extending it with new features...".  The original Transport Tycoon game takes place from 1930 to 2030.  Transport Tycoon Deluxe takes place from 1950 to 2050.  If the intent of OpenTTD is to mimic Transport Tycoon DELUX, inflation should just start at Jan 1st 1950 and then end in 2050.  Disabling inflation should still be an option.  This can allow game scripts/mods/whatever to implement their own forms of inflation for whatever date range they want.

**Comment by jcoletto77** [comment]
In my opinion, one of the issues here is the very questionable decision to inflate costs to a higher extent than revenues.

This is highly unrealistic, to begin with. In the real world there is something called "economies of scale", which allow a large company to be more efficient than a small one. Smaller companies have a limited reach, can't withstand the costs of innovating, and have trouble serving new cohorts of customers. On the other hand, once a company has reached a certain level of fixed costs, it can lever its infrastructure to increase production or spend more in R&D without adding additional overhead. This is how it can afford more advertising, it can broaden its reach, or explore new markets.
And this is exactly the principle that allows giant corporations to remain profitable. If costs actually increased more than revenues, no company whatsoever would be larger than the hole-in-the-wall restaurant across the street. Entrepreneurs are not fools: no one would ever expand their company if they knew they'd face additional costs higher than their additional revenues.
Applied to a transportation company, this means that the cost of adding new infrastructure (new rail lines, new stations, new powerful engines...) does NOT increase more rapidly than the additional income this infrastructure builds. Quite the opposite. A new rail line can be shared; a larger depot can be ran by a similar number of employees; the same manager can lead a team of 2 or 6 employees; maintaining a 1000hp engine does not cost three times as much as maintaining a 500hp one.

I understand that this concept has been introduced with the sole scope of making the game more challenging, but being so unrealistic, its final effect is to make the game just too frustrating.
A good balance between realism and gameplay would be to make costs and revenues increase at the same rate: at most, we could have higher costs for *building* infrastructure, but not for *maintaining* it.

-EDIT- I noticed my nickname doesn't show up here... this is Snail from the TT forums

**Comment by jcoletto77** [comment]
As for inflation, I'm not sure about how it actually works in OTTD, but it seems to me its rate is fixed across the whole duration of the game (please correct me if I'm wrong).
If so, again, this is very unrealistic and it leads to an exponential growth in nominal prices. It would be much better if OTTD randomly varied inflation from year to year, with some persistence (maybe through a random walk). Inflation should go from slightly negative (maybe around -2% yearly, i.e. deflation) to very high (perhaps around 16% yearly: emerging economies, however, experienced much higher levels than that) with random changes of, say, from zero to half a percent per year. The long-term average should be somewhere between 3% and 4% and the distribution should be skewed towards lower levels. The final result should be long periods of low inflation, alternated by rather short stints of very high levels.
We could even add random changes for certain commodities. Coal or oil prices could randomly increase, causing (1) higher maintenance costs of steamers/diesel engines, and at the same time, (2) higher revenues for delivering them as cargoes.
Something like this would help preventing explosion of nominal prices, while making the game more interesting and challenging.

-EDIT- As for the previous message, my nickname doesn't show up. This is Snail from the TT forums

**Reply by 2TallTyler** [reply]
This sounds more like Game Script territory than base game behavior. I don't think any of these are currently exposed to the GS API, but could probably be added.

**Reply by James103** [reply]
> I don't think any of these are currently exposed to the GS API, but could probably be added

JGRPP already allows a Game Script to set the inflation factor to any amount (between 1.0 and `MAX_INFLATION`) at any time, and also allows controlling the inflation factor for costs and payments separately from each other.
Do note even when this is the case, the cost inflation factor is always greater than or equal to the income inflation factor.

See also https://github.com/JGRennison/OpenTTD-patches/commit/1506479f569b8c71e073c68172d761a8a6b1352c, https://github.com/JGRennison/OpenTTD-patches/commit/81291c4ffca92eb78489c261b57135ecef15e775, and https://github.com/JGRennison/OpenTTD-patches/commit/4b9d5a437f9c787425fae55c95330ff916691179
